tt_metal/impl/buffers/buffer.cpp

```diff
@@ -16,6 +16,83 @@ namespace tt {
 
 namespace tt_metal {
 
+#if defined(TRACY_ENABLE)
+//TODO: Because of how tracy mem pool only accepts char const *, we need generate all the literals
+// Need to come up with a compile time trick to generate literals for max number of device
+constexpr int MAX_MEM_DEVICE_SUPPORTED = 4;
+
+static char const *get_memory_pool_type(BufferType buffer_type) {
+    switch (buffer_type) {
+        case BufferType::DRAM: return "DRAM";
+        case BufferType::L1: return "L1";
+        case BufferType::SYSTEM_MEMORY: return "SYSTEM_MEMORY";
+        default: return "UNKNOWN";
+    }
+}
+// Manually generate upto 8 device memory pool literals
+static char const *get_memory_pool_name(BufferType buffer_type, int device_id) {
+    switch (device_id) {
+        case 0:
+            switch (buffer_type) {
+                case BufferType::DRAM: return "DRAM 0";
```

Comment:
can this be simplified w/ magic_enum and string catenation?

Reply:
Yeah, The source of the issue is that tracy mem profiler macro is looking for a `const char *` literal. 
I could not find a neat way to concat the output of magicenum to the runtime variable `device_id`. 
The switch case is basically generating all the literals and switching between them at runtime.

Reply:
We can't do something like
`return std::string(magic_enum::enum_name(buffer_type)) + " " + std::to_string(device_id).c_str();`
or
`return fmt::format("{} {}", magic_enum::enum_name(buffer_type), device_id).c_str();`?

Reply:
No they go out of scope so the `c_str` pointer is released. I get gibberish. Tracy is referring back to these locations after the original call which makes sense, it is not copying it.

Reply:
I see, so there's no way for us to know when Tracy is actually finished/used the string?

Wonder if we can dynamically store these strings in a map that we clear during teardown after we know tracy is done with it?

Reply:
Populating it dynamically at runtime shouldn't be too bad right?

Basically if buffer_type + device id is a new combo, create/insert a new string into the map and pass the reference to it to tracy

Reply:
Hmm yeah I will give it a try we can make it can be a global map that stays alive till the end of program.  but tbh, the api in tracy is not meant to be used like that. Tracy's apis that accept runtime strings, ask for str length as well so that they can do the copy. This api according to tracy's manual is looking for a literal.  

The code here is not pretty but is using the api accordingly.

Reply:
Huh, this seems to be working. Global map populated as buffer locations come up. 

I am still a bit unsure about how string size is being dealt with here. `c_str` is null terminated so probably that is how it is getting determined. 

This feature is not super critical and turned off by default. Let's role with it and if we see any strangeness we can role back